src/core/interfaces.ts

```diff
@@ -476,6 +476,29 @@ export namespace CoreChartProps {
   export type I18nStrings = CartesianI18nStrings & PieI18nStrings;
 }
 
+export interface CoreLegendProps {
+  series: Highcharts.SeriesOptionsType[];
+  title?: string;
+  actions?: React.ReactNode;
+  position?: "bottom" | "side";
+  i18nStrings?: BaseI18nStrings;
+  onClearHighlight?: NonCancelableEventHandler<CoreChartProps.HighlightClearDetail>;
+  onItemHighlight?: NonCancelableEventHandler<CoreChartProps.LegendItemHighlightDetail>;
+  onVisibleItemsChange?: NonCancelableEventHandler<CoreChartProps.VisibleItemsChangeDetail>;
+  getLegendTooltipContent?: CoreChartProps.GetLegendTooltipContent;
+  callback?: (api: CoreLegendProps.LegendAPI) => void;
```

Comment:
This callback-style API is HC-special. The problem is that the initialisation happens not in-sync with React lifecycle, so the ref-based option does not really work. That is not a problem for our standalone legend though. I suggest that we use a ref instead (with imperative handle):

```
const legendApi = useRef(null);

const handler = () => {
  legendApi.current.clearHighlight();
}

return <CoreLegend ref={legendApi} {...otherProps} />
```